### PR TITLE
Accelerate by 36,000x and organize

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
This accelerates the test case by 36,000x and organizes it. The high level scripts are pulled out of `src` and coexist is made into a proper package (note: the name of the module should change because Julia module conventions have a capital in the front!). Then, a code generation scheme via ModelingToolkit is setup as the `dydt_generator.jl`, which results in a 36,000x acceleration of the ODE solve. To demonstrate caching I show that the generated code can be saved to a file and then is benchmarked in `pyjulia_benchmark.jl`. However, note that saving to a file isn't required (as the `dydt_generator.jl` doesn't use the file generated version at all), so generated fast forms can be used on the fly if needed.

Note that ModelingToolkit doesn't seem to be totally needed here, you can get a good portion (but not all) of the speedup by preallocating buffers and rewriting the einsum expressions. However, this just demonstrates a way to use the code generators such that 0 work is required to get the optimal expressions. The function that is generated is in a sparse form so as the networks get larger the amount of acceleration should keep increasing as well.

Note that the only "glut" in the system here is that the symbolic simplification for generating the fastest form of the function takes around 80 seconds. But this is unnecessary since most of the values are zeros. We should open an issue about this an invite @yingboma and @shashi to the repo to take a look at some accelerated simplification tools for this case.

### CODE REVIEW

**Comment by aa25desh** [comment]
Thank you very much! 😄 
So speed up is the effect of the proper package?

**Comment by ChrisRackauckas** [comment]
No, it's from generating a non-allocating dydt.

File: scripts/dydt_generator.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+	# @einsum to_be_modified[l,k,j,i] := trTensor_complete[l,k,j,l,k,j,i]
+	to_be_modified = coexist.einsum("ijkljkl->ijkl", trTensor_complete, T)
+	to_be_modified .-= coexist.einsum("...jkl->...", trTensor_complete, T)
+	# @einsum trTensor_complete[l,k,j,l,k,j,i] = view[l,k,j,i]
+	trTensor_complete = coexist._einsum11(trTensor_complete, to_be_modified, T)
+
+
+	dydt = coexist.einsum("ijkl,ijklmnp->imnp", stateTensor, trTensor_complete, T)
+	return reshape(dydt, prod(size(dydt)))
```
Comment (YingboMa) [inline scripts/dydt_generator.jl:43]:
```suggestion
	return vec(dydt)
```

File: scripts/dydt_generator.jl [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+function solveSystem_unoptimized(
+	state,
+	timeSpan = (0.0, 80.0),
+	p=nothing
+	)
+	prob = ODEProblem(dydt_Complete,state,(0.0,80.0),p=nothing)
+	sol = solve(prob,Tsit5(),reltol=1e-3,abstol=1e-3)
+	sol = convert(Array, sol)
+	return reshape(sol, (4,4,8,9, Int64(prod(size(sol))/(4*4*8*9))))
```
Comment (YingboMa) [inline scripts/dydt_generator.jl:72]:
```suggestion
	return reshape(sol, (4,4,8,9, prod(size(sol)) ÷ (4*4*8*9)))
```

File: scripts/dydt_generator.jl [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+function solveSystem(
+	state,
+	timeSpan = (0.0, 80.0),
+	p=nothing
+	)
+	prob = ODEProblem(f,state,(0.0,80.0),p=nothing)
+	sol = solve(prob,Tsit5(),reltol=1e-3,abstol=1e-3)
+	sol = convert(Array, sol)
+	return reshape(sol, (4,4,8,9, Int64(prod(size(sol))/(4*4*8*9))))
```
Comment (YingboMa) [inline scripts/dydt_generator.jl:83]: same here.

File: scripts/dydt_generator.jl [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
```
Comment (ChrisRackauckas) [inline scripts/dydt_generator.jl:83]:
```suggestion
	reshape(sol, (4,4,8,9, length(sol) ÷ (4*4*8*9)))
```

**Pull request merged**